### DISCUSSION TITLE
VM/370 Build Broken

### DISCUSSION THREAD

**Discussion by adesutherland** [adesutherland/CREXX#294] [General]
The VM/370 Build is broken mainly because compiling the main interpreter source file (rxvmintp.c) causes an out of memory error.

There are a number of solutions
- For VM we can break the file up (although we will need a way to do this smartly so the Windows/Linux/Mac build works as fast as it is now)
- We can cross compile from linux (or whatever)
- We might be able to squeeze it in by unloading BREXX
- S/380 ...
- Some other cunning plan I can't think of right now! 

I think cross compiling may be the way to go.

Anyway - something else for the todo list! I am thinking we need to feature fix CREXX Level B and then fix the port at that point, rather than get side tracked

What do you guys think?

**Comment by mgrossmann** [comment]
I would vote for cross compiling.


Best, Mike


Adrian Sutherland ***@***.***> schrieb am Di. 24. Mai 2022 um
22:06:

> The VM/370 Build is broken mainly because compiling the main interpreter
> source file (rxvmintp.c) causes an out of memory error.
>
> There are a number of solutions
>
>    - For VM we can break the file up (although we will need a way to do
>    this smartly so the Windows/Linux/Mac build works as fast as it is now)
>    - We can cross compile from linux (or whatever)
>    - We might be able to squeeze it in by unloading BREXX
>    - S/380 ...
>    - Some other cunning plan I can't think of right now!
>
> I think cross compiling may be the way to go.
>
> Anyway - something else for the todo list! I am thinking we need to
> feature fix CREXX Level B and then fix the port at that point, rather than
> get side tracked
>
> What do you guys think?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/adesutherland/CREXX/discussions/294>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAF7Q7NWVMADHTTSQIPH4CTVLUZEDANCNFSM5W2XOISQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
Von Gmail Mobile gesendet

**Comment by RexxLA** [comment]
cross compiling is also my preferred solution. Or that GCC might be loaded in overlays ... we might need Bob to look into this for us.
I suspect breaking up the source will not make a great difference for the other platforms, so that might be the short term stopgap.

**Comment by BobBolch** [comment]
I looked into the 380 stuff a few months ago, but, it looks like you need a
special Hercules setup with a very old unsupported Hercules version. All
the GCC assembler subroutines are 24 bit, so cross compiling sounds like
the best approach, if that is possible.

I will think about it dome more.
Bob
